### PR TITLE
[TF FE] Add complex type support for AddN

### PR DESCRIPTION
**Pull request by dnabanita7**
### Details:
 - Added Complex type for Addn
 - Added tests to confirm the same

### Tickets:
 - #22944

### CODE REVIEW

File: src/frontends/tensorflow_common/src/op/addN.cpp [hunk 1]
```diff
@@ -18,15 +18,28 @@ OutputVector translate_add_n_op(const NodeContext& node) {
     default_op_checks(node, 1, {"AddN", "ADD_N"});
     int num_size = static_cast<int>(node.get_input_size());
 
-    Output<Node> result = node.get_input(0);
+
+    auto result = node.get_input(0);
     for (int ind = 1; ind < num_size; ++ind) {
         result = make_shared<v1::Add>(result, node.get_input(ind));
     }
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/addN.cpp:25]: please think how you will add operands if they have `ComplexTypeMark` under. Is it valid?

File: src/frontends/tensorflow_common/src/op/addN.cpp [hunk 2]
```diff
@@ -18,15 +18,28 @@ OutputVector translate_add_n_op(const NodeContext& node) {
...
+    auto complex_type_mark_result = as_type_ptr<ComplexTypeMark>(result.get_node_shared_ptr());
+    if (complex_type_mark_result) {
+        FRONT_END_GENERAL_CHECK(complex_type_mark_result != nullptr,
+                                "AddN/ADD_N got complex and non-complex inputs. Inputs should be of same type.");
+        result = complex_type_mark_result->input_value(0);
+
+        element::Type complex_part_type_result = complex_type_mark_result->get_complex_part_type();
+        auto complex_result = make_shared<ComplexTypeMark>(result->output(0), complex_part_type_result);
+        return {complex_result};
+    }
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/addN.cpp:36]: How about to handle `ComplexTypeMark` before each operand?

**Comment by rkazants** [comment]
Hi @dnabanita7,

Please re-think about your solution and read one more time how OV supports complex type tensor and what `ComplexTypeMark` means.

Best regards,
Roman

**Comment by rkazants** [comment]
Hi @dnabanita7, any update?

**Comment by rkazants** [comment]
Hi @dnabanita7, any update on this PR? Please be aware that I will have to release GFI of no update for long time.

Best regards,
Roman

**Comment by dnabanita7** [comment]
Hello,

My bad! I haven't been working on the same.

Regards,

**Comment by mlukasze** [comment]
> Hello, My bad! I haven't been working on the same. Regards,

it's ok, but will you have time to finish this task? :)

**Comment by dnabanita7** [comment]
No, I don't have time to finish it.

**Comment by rkazants** [comment]
@dnabanita7, please continue this PR if you have time. For now I release https://github.com/openvinotoolkit/openvino/issues/22944 for other contributors.

Best regards,
Roman

**Comment by rkazants** [comment]
GFI was resolved by https://github.com/openvinotoolkit/openvino/pull/23614

**Pull request closed without merging**